apps/explorer/src/components/SyntaxHighlighter/index.tsx

```diff
@@ -10,14 +12,26 @@ interface Props {
 	code: string;
 	language: Language;
 }
-
+const MAX_LINES = 500;
+// Use scroll to load more lines of code to prevent performance issues with large code blocks
 export function SyntaxHighlighter({ code, language }: Props) {
+	const observerElem = useRef<HTMLDivElement | null>(null);
+	const { isIntersecting } = useOnScreen(observerElem);
+	const [loadedLines, setLoadedLines] = useState(MAX_LINES);
+	useEffect(() => {
+		if (isIntersecting) {
+			loadMoreLines();
+		}
+	}, [isIntersecting]);
+	const loadMoreLines = () => {
```

Comment:
this is probably not necessary right? We could move the `setLoadedLines` in the `useEffect`